src/entities/todos/model/todo.ts

```diff
@@ -1,12 +1,48 @@
 import {createSlice} from "@reduxjs/toolkit";
-import {ITodo} from "../ui/todo";
+import {ITodo} from "../../../shared/interfaces";
 
-const initialState: ITodo[] = []
+interface IInitialState {
+    todos: ITodo[]
+}
+
+const initialState:IInitialState = {
+    todos: []
+}
+
+const findTaskById = (state:IInitialState, id: number) => state.todos.find(task => task.id === id)
 
 const todosSlice = createSlice({
     name: 'todos',
     initialState,
-    reducers: {}
+    reducers: {
+        addNewTask: (state, action) => {
+            state.todos.push(action.payload)
```

Comment:
return { // returning a copy of orignal state 
    ...state, //spreading the original state
    todos: [action.payload, ...state.todos] // new todos array
   }